src/dune_engine/build_system.ml

```diff
@@ -12,9 +12,8 @@ module Fs : sig
   val assert_exists : loc:Loc.t -> Path.t -> unit Memo.Build.t
 end = struct
   let mkdir_p_def =
-    Memo.create "mkdir_p"
+    Memo.create_no_cutoff "mkdir_p"
```

Comment:
Maybe worth leaving a comment if no_cutoff is in fact necessary (and whether it's sufficient). (it's actually not clear to me how to think about when mkdir needs to be invalidated; maybe we need to invalidate it when we delete stale dirs, but how can we arrange that invalidation? no_cutoff probably isn't going to help with that)

Reply:
Sorry, I thought you changed it from cutoff to no cutoff, but you didn't. Never mind. (it's still something that bothers me, but your change has nothing to do with it)

Reply:
Indeed. I added a comment.